### PR TITLE
Feature/ftp source

### PR DESCRIPTION
**Pull request by gokulavasan**
Reopening PR https://github.com/caskdata/hydrator-plugins/pull/175

FTP and SFTP batch source PR.

### CODE REVIEW

File: core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/FTPBatchSource.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    @Description("Regex to filter out filenames in the path. Defaults to '.*'")
+    @Nullable
+    public String fileRegex;
+
+    @Description(FileBatchSource.INPUT_FORMAT_CLASS_DESCRIPTION)
+    @Nullable
+    public String inputFormatClassName;
+
+    public FTPBatchSourceConfig() {
+      this.fileSystemProperties = GSON.toJson(ImmutableMap.<String, String>of());
```
Comment (gsps1) [inline core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/FTPBatchSource.java]: serializing empty map looks weird, do we ned this?

Comment (gsps1) [inline core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/FTPBatchSource.java]: also constructors are used only in tests right?

File: core-plugins/src/test/java/co/cask/hydrator/plugin/batch/ETLFTPTestRun.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+  @After
+  public void stop() throws Exception {
+    if (ftpServer != null) {
+      ftpServer.stop();
+    }
+  }
+
+  // TODO: In unit-test, the classpath has hadoop-common jar ahead of cdap-app-fabric and hence the modified
+  // FTPInputStream.java in cdap-app-fabric is not getting picked up. Hence ignoring the test.
```
Comment (gsps1) [inline core-plugins/src/test/java/co/cask/hydrator/plugin/batch/ETLFTPTestRun.java:98]: create a JIRA about this issue and what's needed for the fix and link it here.

**Comment by gsps1** [comment]
just one comment about creating JIRA, LGTM otherwise.

**Comment by gokulavasan** [comment]
Added JIRA link. Merging.

**Pull request merged**